src/main/play-doc/operation/DynamicProxyConfiguration.md

```diff
@@ -0,0 +1,585 @@
+
+[[ConductR CLI|CLI]] is required to perform the following steps, and as such it needs to be installed beforehand.
+
+These are the steps to deploying custom HAProxy configuration.
+
+### Setup
+
+Create a directory where the custom HAProxy configuration will be placed, e.g.
+
+```lang-none
+mkdir -p /tmp/custom-haproxy-conf
+```
+
+### Populate the custom HAProxy configuration template
+
+Create the custom HAProxy template, e.g.
+
+```lang-none
+touch /tmp/custom-haproxy-conf/haproxy-override.cfg
+```
+
+The file can be given any name as long as it's within the directory created from the previous step.
+
+In this example, we'll populate the file with the following custom template.
+
+```
+# HAProxy Specific Configuration
+global
+	log /dev/log	local0
+	log /dev/log	local1 notice
+	chroot /var/lib/haproxy
+	stats socket /run/haproxy/admin.sock mode 660 level admin
+	stats timeout 30s
+	user haproxy
+	group haproxy
+	daemon
+
+defaults
+	log	global
+	mode	http
+	option	httplog
+	option	dontlognull
+    timeout connect 5000
+    timeout client  50000
+    timeout server  50000
+
+
+# Custom Endpoint Configuration
+{{#eachAcls bundles defaultHttpPort=9443}}
+  {{#ifAcl 'public-site' '1' 'main'}}
+frontend my_www_frontend
+  bind {{haproxyHost}}:80
+  # Uncomment the following line if SSL is used
+  # bind {{haproxyHost}}:443 ssl crt {{overrideConfigDir}}/test-cert.pem
+  mode http
+  acl my_www_host_match hdr(host) -i www.acme.com
+  use_backend my_www_backend if my_www_host_match
+
+backend my_www_backend
+  mode http
+    {{#eachBackendServer}}
+  server {{serverName}} {{host}}:{{port}} maxconn 1024
+    {{/eachBackendServer}}
+  {{/ifAcl}}
+{{/eachAcls}}
+```
+
+Here are the explanation for the example template above:
+
+* *HAProxy Specific Configuration* section is declared prior *Custom Endpoint Configuration* section.
+* *HAProxy Specific Configuration* section is populated with operating system specific setup. In the example above it is populated with a cut-down version of Ubuntu's configuration.
+* *Custom Endpoint Configuration* section contains application specific proxy configuration.
+* The endpoint that matches the following criteria has a custom configuration declared within the [ifAcl](#ifAcl) block:
+    * `system` is `public-site`
+    * `system version` is `1`
+    * `endpoint name` is `main`
+* HAProxy frontend `my_www_frontend` is configured to listen on HTTP port `80` and will relay request for `www.acme.com` to `my_www_backend`.
+* HAProxy backend `my_www_backend` server is generated based on the backend server information contained within the [eachBackendServer](#eachBackendServer) helper function.
+* The backend server information will be updated whenever there's changes to the endpoint, e.g. bundle instances being started or stopped the within the ConductR cluster.
+* All other endpoints is for configured using ConductR default configuration. The HTTP endpoint is exposed on the port `9443`
```

Comment:
@markusjura - remove the confusing remarks for the 4 bullet points above - hopefully it's clear the way it is now.